php/commands/src/CLI_Command.php

```diff
@@ -179,35 +179,158 @@ public function info( $_, $assoc_args ) {
+				EE::success( "EasyEngine is at the latest{$update_type}version." );
+				return;
+			}
+			$newest = $updates[0];
+			EE::confirm( sprintf( 'You have version %s. Would you like to update to %s?', EE_VERSION, $newest['version'] ), $assoc_args );
+			$download_url = $newest['package_url'];
+			$md5_url = str_replace( '.phar', '.phar.md5', $download_url );
 		}
-
-		$config_file = fopen( $config_file_path, "w" );
-		fwrite( $config_file, Spyc::YAMLDump( $existing_config ) );
-		fclose( $config_file );
-
-		file_put_contents( EE_CONF_ROOT . '/update.sh', file_get_contents( 'http://rt.cx/eev4' ) );
-		if ( \EE\Utils\default_launch( 'bash ' . EE_CONF_ROOT . '/update.sh' ) ) {
-			EE::success( 'Update complete.' );
-			unlink( EE_CONF_ROOT . '/update.sh' );
+		EE::log( sprintf( 'Downloading from %s...', $download_url ) );
+		$temp = \EE\Utils\get_temp_dir() . uniqid( 'ee_', true ) . '.phar';
+		$headers = array();
+		$options = array(
+			'timeout' => 600,  // 10 minutes ought to be enough for everybody.
+			'filename' => $temp,
+		);
+		Utils\http_request( 'GET', $download_url, null, $headers, $options );
+		$md5_response = Utils\http_request( 'GET', $md5_url );
+		if ( 20 != substr( $md5_response->status_code, 0, 2 ) ) {
+			EE::error( "Couldn't access md5 hash for release (HTTP code {$md5_response->status_code})." );
+		}
+		$md5_file = md5_file( $temp );
+		$release_hash = trim( $md5_response->body );
+		if ( $md5_file === $release_hash ) {
+			EE::log( 'md5 hash verified: ' . $release_hash );
 		} else {
-			EE::error( 'There was some error in running update. Please check logs and re-run update.' );
+			EE::error( "md5 hash for download ({$md5_file}) is different than the release hash ({$release_hash})." );
+		}
+		$php_binary = Utils\get_php_binary();
+		$process = EE\Process::create( "{$php_binary} $temp cli info" );
+		$result = $process->run();
+		if ( 0 !== $result->return_code || false === stripos( $result->stdout, 'EE version' ) ) {
+			$multi_line = explode( PHP_EOL, $result->stderr );
+			EE::error_multi_line( $multi_line );
+			EE::error( 'The downloaded PHAR is broken, try running ee cli update again.' );
+		}
+		EE::log( 'New version works. Proceeding to replace.' );
+		$mode = fileperms( $old_phar ) & 511;
+		if ( false === chmod( $temp, $mode ) ) {
+			EE::error( sprintf( 'Cannot chmod %s.', $temp ) );
+		}
+		class_exists( '\cli\Colors' ); // This autoloads \cli\Colors - after we move the file we no longer have access to this class.
+		if ( false === rename( $temp, $old_phar ) ) {
+			EE::error( sprintf( 'Cannot move %s to %s', $temp, $old_phar ) );
+		}
+		if ( Utils\get_flag_value( $assoc_args, 'nightly' ) ) {
+			$updated_version = 'the latest nightly release';
+		} elseif ( Utils\get_flag_value( $assoc_args, 'stable' ) ) {
+			$updated_version = 'the latest stable release';
+		} else {
+			$updated_version = $newest['version'];
+		}
+		EE::success( sprintf( 'Updated WP-CLI to %s.', $updated_version ) );
+	}
+
+		/**
```

Comment:
Following phpcs wherever possible would be preferred.

Reply:
Done.